### PR TITLE
Space dims fix

### PR DESCRIPTION
**Pull request by navjotk**
This PR adds a test that exposes the bug related to setting the `space_dims` parameter in the Propagator. 
It also includes a fix for the bug.

### CODE REVIEW

File: tests/test_propagator.py [hunk 1]
```diff
@@ -44,6 +44,16 @@ def test_4d(self):
         f(data, arr)
         assert(arr[4][3][2][1] == 122)
 
+    def test_space_dims_2d(self):
+        space_dims = (z, x)
+        propagator = Propagator("process", 1, (4, 3), [], space_dims=space_dims)
```
Comment (mlange05) [inline tests/test_propagator.py]: This looks like it only tests prescribed `space_dims`. Should we also test the defaulting behaviour?

**Comment by navjotk** [comment]
The test failing now is the due to the bug that is exposed in the code that guesses the dimensions at  https://github.com/opesci/devito/blob/master/devito/operator.py#L211

**Comment by mlange05** [comment]
Ok, I fixed the dimension guess bug by not using sets and thus enforcing dimension index ordering. If I can get some confirmation that the fix looks ok, I think we can merge this.

**Comment by FabioLuporini** [comment]
Looks reasonable

**Comment by mlange05** [comment]
Great, thanks. Merging...

**Pull request merged**